testing/e2e/e2e_api_test.go

```diff
@@ -48,3 +57,191 @@ func (s *BeaconKitE2ESuite) TestBeaconAPIStartup() {
 	s.Require().NotEmpty(stateRootResp)
 	s.Require().False(stateRootResp.Data.IsZero())
 }
+
+// TestBeaconValidatorsWithIndices tests the beacon node api for beacon validators with indices.
+func (s *BeaconKitE2ESuite) TestBeaconValidatorsWithIndices() {
+	client := s.initBeaconTest()
+
+	indices := []phase0.ValidatorIndex{0}
+	validatorsResp, err := client.Validators(
+		s.Ctx(),
+		&beaconapi.ValidatorsOpts{
+			State:   utils.StateIDHead,
+			Indices: indices,
+		},
+	)
+	s.Require().NoError(err)
+	s.Require().NotNil(validatorsResp)
+
+	validatorData := validatorsResp.Data
+	s.Require().NotNil(validatorData, "Validator data should not be nil")
```

Comment:
these two checks are the same. We should drop one of them

Reply:
I would say checking both explicitly is fine

Reply:
In one, we are checking the response and in another we are checking Data inside the response.

Reply:
You're right, I got confused here